### PR TITLE
Cluster manager

### PR DESCRIPTION
**Pull request by kornys**
### Type of change

- Enhancement / new feature

### Description

management of kube cluster, operations like scale nodes, 

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by kornys** [comment]
@lulf maybe I can merge it with KubeCMDClient and make all methods static instead of manager as singleton, WDYT?

**Comment by kornys** [comment]
@enmasse-ci run tests profile=smoke crc test_only

File: systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Do the operation with cluster (scale, restart etc...)
+ */
+public class KubeClusterManager {
```
Comment (lulf) [inline systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java:25]: Maybe rename OpenShiftClusterManager? I think almost everything here is openshift-specific.

File: systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    //////////////////////////////////////////////////////////////
+    public void drainNode(String nodeName) {
+        LOGGER.info("Cluster node {} is going to drain", nodeName);
+        setNodeSchedule(nodeName, false);
+        Exec.execute(Arrays.asList(CMD, "adm", "drain", nodeName, "--delete-local-data", "--ignore-daemonsets"));
+    }
+
+    public void setNodeSchedule(String node, boolean schedule) {
+        LOGGER.info("Set {} schedule {}", node, schedule);
+        Exec.execute(Arrays.asList(CMD, "adm", schedule ? "uncordon" : "cordon", node));
```
Comment (lulf) [inline systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java:94]: Check for errors?

File: systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        }, new TimeoutBudget(20, TimeUnit.MINUTES));
+    }
+
+    //////////////////////////////////////////////////////////////
+    // Evacuate pods from nodes
+    //////////////////////////////////////////////////////////////
+    public void drainNode(String nodeName) {
+        LOGGER.info("Cluster node {} is going to drain", nodeName);
+        setNodeSchedule(nodeName, false);
+        Exec.execute(Arrays.asList(CMD, "adm", "drain", nodeName, "--delete-local-data", "--ignore-daemonsets"));
```
Comment (lulf) [inline systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java:89]: Check for errors?

File: systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                node.getMetadata().getName().contains("master")).collect(Collectors.toList());
+    }
+
+    //////////////////////////////////////////////////////////////
+    // Scale openshift cluster
+    //////////////////////////////////////////////////////////////
+    public void scaleOpenshiftWorkers(int workerCount) {
+        LOGGER.info("Cluster is going to be scaled to {} worker nodes", workerCount);
+        String machineSetName = Exec.execute(Arrays.asList(CMD, "get", "machinesets", "-n", "openshift-machine-api", "-o", "jsonpath='{.items[0].metadata.name}'")).getTrimmedStdOut();
+        Exec.execute(Arrays.asList(CMD, "scale", "--replicas", Integer.toString(workerCount), "machineset", machineSetName, "-n", "openshift-machine-api"));
```
Comment (lulf) [inline systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java:73]: Check for errors?

File: systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java (hunk 1 above)
Comment (kornys) [reply systemtests/src/main/java/io/enmasse/systemtest/platform/cluster/KubeClusterManager.java:25]: @lulf only worker scale is ocp specific, node drain is kubernetes stuff as well

**Comment by kornys** [comment]
@enmasse-ci run tests profile=smoke crc test_only

**Pull request merged**